master/buildbot/process/botmaster.py

```diff
@@ -28,6 +28,54 @@
 from buildbot.util import service
 
 
+class BotLockTracker:
+    ''' Tracks real lock instances given lock ID and config_version. The latter
+        is used as a way to track most recent properties of real locks.
+
+        This approach is needed because there's no central registry of locks
```

Comment:
Central registry? this looks like a job for the SharedService manager.

Can we implement this as a sharedservice?

http://docs.buildbot.net/latest/developer/utils.html#buildbot.util.service.SharedService

each Lock would become a SharedService.